src/status_im2/navigation/core.cljs

```diff
@@ -1,88 +1,110 @@
+
+(navigation/set-lazy-component-registrator
+ (fn [key]
+   (let [screen (views/screen key)]
+     (navigation/register-component key
+                                    (fn [] (gesture/gesture-handler-root-hoc screen))
+                                    (fn [] screen)))))
+
+;; APP LAUNCHED
+(navigation/reg-app-launched-listener
+ (fn []
+   (reset! state/curr-modal false)
+   (reset! state/dissmissing false)
+   (if (= @state/root-id :multiaccounts-stack)
+     (re-frame/dispatch-sync [:set-multiaccount-root])
+     (when @state/root-id
+       (navigation/set-root (get (roots/roots) @state/root-id))
+       (re-frame/dispatch [::login-core/check-last-chat])))
+   (rn/hide-splash-screen)))
+
+(defn set-view-id
+  [view-id]
+  (when-let [{:keys [on-focus]} (get views/screens view-id)]
+    (re-frame/dispatch [:set-view-id view-id])
+    (re-frame/dispatch [:screens/on-will-focus view-id])
+    (when on-focus
+      (re-frame/dispatch on-focus))))
+
+(navigation/reg-component-did-appear-listener
+ (fn [view-id]
+   (when (get views/screens view-id)
+     (set-view-id view-id)
+     (when-not @state/curr-modal
+       (reset! state/pushed-screen-id view-id)))))
+
+(defn dissmissModal
+  []
+  (reset! state/dissmissing true)
+  (navigation/dismiss-modal (name (last @state/modals))))
 
 (defn dismiss-all-modals
   []
-  (log/debug "dissmiss-all-modals"
-             {:curr-modal  @state/curr-modal
-              :modals      @state/modals
-              :dissmissing @state/dissmissing})
   (when @state/curr-modal
     (reset! state/curr-modal false)
     (reset! state/dissmissing true)
     (doseq [modal @state/modals]
       (navigation/dismiss-modal (name modal)))
     (reset! state/modals [])))
 
-(defn status-bar-options
-  []
-  (if platform/android?
-    {:navigationBar {:backgroundColor (colors/theme-colors colors/white colors/neutral-100)}
-     :statusBar     {:translucent     true
-                     :backgroundColor :transparent
-                     :drawBehind      true
-                     :style           (if (colors/dark?) :light :dark)}}
-    {:statusBar {:style (if (colors/dark?) :light :dark)}}))
-
-;; PUSH SCREEN TO THE CURRENT STACK
+;; ROOT
+(re-frame/reg-fx
+ :set-root
+ (fn [root-id]
+   (let [root (get (roots/roots) root-id)]
+     (dismiss-all-modals)
+     (re-frame/dispatch [:multiaccounts.ui/switch-theme
+                         (get roots/themes root-id)
+                         root-id])
+     (reset! state/root-id (or (get-in root [:root :stack :id]) root-id))
+     (navigation/set-root root))))
+
+;; NAVIGATE-TO
 (defn navigate
   [comp]
-  (log/debug "NAVIGATE" comp)
   (let [{:keys [options]} (get views/screens comp)]
+    (dismiss-all-modals)
     (navigation/push
      (name @state/root-id)
      {:component {:id      comp
                   :name    comp
-                  :options (merge (status-bar-options)
+                  :options (merge (options/statusbar-and-navbar)
```

Comment:
Consider removing any option merging logic to a helper function in options.cljs.
As I see it we have 3 different levels of options.
Default options (defined in options.cljs), root level options (Defined on the root stack in roots.cljs) and screen options.
This function here should not have any options directly being set (e.g (if topBar options etc....)
Instead it should just do a nested merge respecting this 3 tier hierarchy for the different options. 


```
  {:component {:id      comp
                  :name    comp
                  :options (merge-options (options/default-root)
                                                              (roots/get-stack-options stack-id) ;;
                                                               options)
 ```
Where this merge-options function has to be created but better to have the logic abstracted outside this navigate function as we might need to use in other places as well.

Reply:
also if part of this code is about managing the old screens vs new screens, we can have two functions by this and add a boolean property `old-ui` to the old screens and then use that here.
```
(let [{:keys [options old-ui]} (get views/screens comp)]
(dismiss-all-modals)
    (navigation/push
     (name @state/root-id)
     {:component {:id      comp
                  :name    comp
                  :options (if old-ui? (merge-old-options options ...) (merge-options ...)
```

Reply:
thanks, yeah i would do it in the separate PR, still lots of work to do, this PR is already huge but its only first iteration